internal/utilities/password.go

```diff
@@ -0,0 +1,65 @@
+package utilities
+
+import (
+	"crypto/rand"
+	"math/big"
+	"strings"
+)
+
+// parseGroups processes the required character groups from a slice of strings.
+func parseGroups(requiredChars []string) []string {
+	var groups []string
+	groups = append(groups, requiredChars...)
+	return groups
+}
+
+func GeneratePassword(requiredChars []string, length int) (string, error) {
+	groups := parseGroups(requiredChars)
```

Comment:
I think we can reference the parameter directly since we don't mutate the slice.
```suggestion
func GeneratePassword(requiredChars []string, length int) (string, error) {
```